js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx

```diff
@@ -102,11 +102,40 @@ export const AssetNodeOverview = ({
     return <AssetNodeOverviewLoading />;
   }
 
-  let tableSchema = materialization?.metadataEntries.find(isCanonicalTableSchemaEntry);
-  let tableSchemaLoadTimestamp = materialization ? Number(materialization.timestamp) : undefined;
-  if (!tableSchema) {
-    tableSchema = assetNode?.metadataEntries.find(isCanonicalTableSchemaEntry);
-    tableSchemaLoadTimestamp = assetNodeLoadTimestamp;
+  const materializationTableSchema = materialization?.metadataEntries.find(
+    isCanonicalTableSchemaEntry,
+  );
+  const materializationTableSchemaLoadTimestamp = materialization
+    ? Number(materialization.timestamp)
+    : undefined;
+  const definitionTableSchema = assetNode?.metadataEntries.find(isCanonicalTableSchemaEntry);
+  const definitionTableSchemaLoadTimestamp = assetNodeLoadTimestamp;
+
+  let tableSchema = materializationTableSchema ?? definitionTableSchema;
+  const tableSchemaLoadTimestamp =
+    materializationTableSchemaLoadTimestamp ?? definitionTableSchemaLoadTimestamp;
+
+  // Merge the descriptions from the definition table schema with the materialization table schema
+  if (materializationTableSchema && definitionTableSchema) {
+    const definitionTableSchemaColumnDescriptionsByName =
+      definitionTableSchema.schema.columns.reduce(
```

Comment:
This all looks great -  there's a super neat new way to do this in modern JS using `Object.fromEntries`, which turns an array of [key, value] entries into the object of ({keys: values}) without you needing to specify the types again on line 128

I think this would be:

```
const definitionTableSchemaColumnDescriptionsByName = Object.fromEntries(      definitionTableSchema.schema.columns.map(col => [col.name, col.description])
)
```

It seems like the name && description check isn't necessary because we check again for description on 132